src/protocol/vm/state.rs

```diff
@@ -875,6 +879,38 @@ mod tests {
             .clone()
             .ensure_capability(Capability::MarginalPrice)
             .is_err());
+
+        // Verify all tokens are initialized in the engine
+        let engine_accounts = pool_state
+            .engine
+            .unwrap()
+            .state
+            .inner
+            .read()
+            .await
+            .clone()
+            .accounts
+            .accounts;
+        for token in pool_state.tokens.clone() {
```

Comment:
As far as I can tell this is the only reason why you exposed the `AccountState`? If so I'd say we create a getter function for it that would be `#[cfg(test)]`

AFAIK it's good practice to keep things as private as possible to avoid people doing unexpected things

Reply:
great idea - thank you!